### PR TITLE
Integrate `remote-pdb` to debug batou runs.

### LINKED ISSUE #149: document how to debug deployment problems
A dedicated page in the docs about debugging **batou** would be nice.

So far I have these experiences...

- you can use `self.log("output")` to print something to the terminal
- you can't set a `breakpoint()` as the deployment breaks without stopping at the breakpoint
  - describe why this happens (because of threading?)
  - alternatives? (except printing to the console?)
- you can use the `-d` flag which enables debugging output during deployment
- ....

### LINKED ISSUE #199: Debugging batou?
To debug issues in `batou` I am nowadays using [remote-pdb](https://pypi.org/project/remote-pdb/). It allows to start a debugger at any point in the code as it opens a remote port where the debugger can be accessed via `nc` or `telnet`.

To be able to use it in the tests I added `remote-pdb` locally to `tox.ini` as a dependency.

Would it be a good idea to add it (or another remote debugger) there permanently? (It does not hurt and eases debugging end to end tests.)
I can live with being forced to have to add the debugger to `requirements.txt` to debug actual `batou` runs.

### PR DESCRIPTION
**Pull request by icemac**
Fixes #199.
Fixes #149.

### CODE REVIEW

File: doc/source/user/advanced.txt [hunk 1]
```diff
@@ -5,6 +5,33 @@ Advanced Usage
 Writing a custom component (TODO)
 ---------------------------------
 
+Debugging batou runs
```
Comment (ctheune) [inline doc/source/user/advanced.txt:8]: Thanks for getting this documented. I'd like to see a few more details:

* how do you exactly configure the environment variables? is it sufficient to call batou locally with those or do we have to do extra work to get them to the remote side? even if nothing special needs to happen, I'd love an explicit example. including an example shell session how this looks like.
* how do you call the debugger on Python 3.6?
* any chance to cover this with some testing?

**Review by ctheune — CHANGES_REQUESTED** [review]

File: doc/source/user/advanced.txt (hunk 1 above)
Comment (icemac) [reply doc/source/user/advanced.txt:8]: * Environment variables are currently not propagated to the remote host. I do not know how to do this.
* I added an example debugging session to the documentation.
* I documented how to use the debugger in Python 3.6.
* I added a simple test to check the default configuration. Testing the usage of the debugger I see no chance.

**Comment by icemac** [comment]
@ctheune Do you agree to the changes I made, so this PR can be merged?

**Comment by ctheune** [comment]
Ok, so forwarding the environment variables would be relatively easy: see the `setup_deployment` function in `remote_core` and you could pass an additional "os_env" argument that would be used to update the os.env on the remote side with selected, known keys (and their values).

**Review by ctheune — CHANGES_REQUESTED** [review]
we can easily propagate the environment, see my standalone comment a few minutes ago.

**Comment by icemac** [comment]
@ctheune I implemented the propagation of the debugging environment variables to the remote host and tested it in a project of ours, that it actually works. – It seems there are no tests for testing an actual remote deployment.

I think this PR is now ready to be merged.

**Pull request merged**